From 2f70cc106df7a0a3ec5d671fd21d2c7e4113504a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Thalheim?= <joerg@thalheim.io>
Date: Mon, 31 Jan 2022 17:43:47 +0100
Subject: [PATCH] x86_64: fix glibc isa detection
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jörg Thalheim <joerg@thalheim.io>
---
 x86/cpuid.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/x86/cpuid.c b/x86/cpuid.c
index c3b67d9..4e26c13 100644
--- a/x86/cpuid.c
+++ b/x86/cpuid.c
@@ -22,10 +22,10 @@ static void filter_cpuid(struct kvm_cpuid2 *kvm_cpuid)
 		switch (entry->function) {
 		case 0:
 			/* Vendor name */
-			memcpy(signature, "LKVMLKVMLKVM", 12);
+			memcpy(signature, "GenuineIntel", 12);
 			entry->ebx = signature[0];
-			entry->ecx = signature[1];
-			entry->edx = signature[2];
+			entry->edx = signature[1];
+			entry->ecx = signature[2];
 			break;
 		case 1:
 			/* Set X86_FEATURE_HYPERVISOR */
-- 
2.34.1

